### ISSUE TITLE
Inputs missing labels & Labels missing inputs

### ISSUE THREAD

**Issue by danspratling** [jackdomleo7/Checka11y.css#7]
## Proposal:
Inputs missing labels (often when Placeholders are being used as labels) might have unclear intentions.

## Issue:
```
<input type="text" placeholder="Email" />
```
The placeholder in the above example is rarely read by screenreaders

## Issue 2:
```
<label>Email</label>
<input type="text" />
```
The label in the above example exists, but is not associated with an element so will often be misinterpreted.

## Solution:
```
<label for="email">Email Address</label>
<input name="email" type="text" />
```
The above is the best-accepted solution, which most screen readers understand (unlike `<label><input></label>`).

We should check 2 things
- [] Does a label have a for attribute present and populated
- [] Does an input have a name attribute present and populated

This won't account for if any inputs/labels have attributes present but their counterparts don't exist, however.

**Comment by jackdomleo7** [comment]
I like this. But what about this scenario, which I often do:

```html
<label>
  <span>Email</span>
  <input type="email" />
</label>
```

I think the above solution would highlight this as an error? 🤔

I have to admit, my knowledge and research on this method in terms of accessibility where the `<label>` wraps the `<input>` is not solid.

**Comment by danspratling** [comment]
It would, but I think as someone suggested to us on twitter, this approach is flawed as some screen readers don't correctly associate the inner input with the wrapping label.

So maybe in this case we would show a warning?

```
Warning: some screen readers don't correctly understand labels not directly associated with inputs using for/name attributes. It is recommended adjusting the approach
```

**Comment by jackdomleo7** [comment]
Ok, that sounds good. I think we should look deeper into #4 first so this works effectively.

**Comment by Being-Maverick** [comment]
Hey @danspratling and @jackdomleo7 , I would like to resolve this issue. Please assign it to me.

**Comment by Lewiscowles1986** [comment]
I am so sorry for tagging this. It was a mistake #73 does not address this in any way

**Comment by jackdomleo7** [comment]
I merged #43 into branch _valid_input_and_label_ and did a little more work.

It was throwing a false error for:
```html
<label>
  Label
  <input />
</label>
```

And I don't think `name` is required on an `<input>`. It's only required if you want to send data to the server.

Either needs a lot more thought or accept we can't do it with CSS.

**Comment by Tushar8466** [comment]
you can assign this to me
